Src/ILGPU/IR/Values/Compare.cs

```diff
@@ -273,6 +274,13 @@ public sealed class CompareValue : Value
                 Kind,
                 Flags);
 
+        /// <summary cref="Value.Write(IIRWriter)"/>
+        protected internal override void Write(IIRWriter writer)
+        {
+            writer.Write("Kind", Kind);
```

Comment:
Can we convert `"Kind"` to `nameof(Kind)` to make it easier to maintain the different named tags?